### PR TITLE
Move line numbers of the Arabic pane to the right

### LINKED ISSUE #327: Move line numbers in arabic pane to right of screen
This may require modifying the `TextLineNumber` Java class.

### PR DESCRIPTION
**Pull request by giordano**
Fix #327.

Preview of the result:

image

### CODE REVIEW

File: src/main/java/uk/ac/ucl/rc/development/oracc/ext/TextLineNumber.java [hunk 1]
```diff
@@ -229,7 +247,6 @@ private void setPreferredWidth()
 			Dimension d = getPreferredSize();
 			d.setSize(preferredWidth, HEIGHT);
 			setPreferredSize( d );
-			setSize( d );
```
Comment (ageorgou) [inline src/main/java/uk/ac/ucl/rc/development/oracc/ext/TextLineNumber.java]: How come this was removed? Did it cause a problem or do nothing?

File: src/main/java/uk/ac/ucl/rc/development/oracc/ext/TextLineNumber.java [hunk 2]
```diff
@@ -61,28 +63,44 @@
...
 		this.component = component;
+		if (left_to_right)
+		{
+			setDigitAlignment( RIGHT );
+			this.OUTER = new MatteBorder(0, 0, 0, 2, Color.GRAY);
+		}
+		else
+		{
+			setDigitAlignment( LEFT );
+			this.OUTER = new MatteBorder(0, 2, 0, 0, Color.GRAY);
```
Comment (ageorgou) [inline src/main/java/uk/ac/ucl/rc/development/oracc/ext/TextLineNumber.java]: Just to understand - the "directions" (left, right, etc) in the MatteBorder constructor are absolute, right? They don't depend on what the orientation of the containing component is?

File: src/main/java/uk/ac/ucl/rc/development/oracc/ext/TextLineNumber.java [hunk 3]
```diff
@@ -61,28 +63,44 @@
...
 	{
-		this(component, 3);
+		this(component, left_to_right, 3);
 	}
 
 	/**
 	 *	Create a line number component for a text component.
 	 *
 	 *  @param component  the related text component
+	 *  @param left_to_right  whether the text is left to right, this
```
Comment (ageorgou) [inline src/main/java/uk/ac/ucl/rc/development/oracc/ext/TextLineNumber.java]:
```suggestion
	 *  @param left_to_right  whether the text is left to right; this
```

File: src/main/java/uk/ac/ucl/rc/development/oracc/ext/TextLineNumber.java (hunk 1 above)
Comment (giordano) [reply src/main/java/uk/ac/ucl/rc/development/oracc/ext/TextLineNumber.java]: If I remember correctly, it was doing nothing.  See https://stackoverflow.com/q/1783793/2442087

File: src/main/java/uk/ac/ucl/rc/development/oracc/ext/TextLineNumber.java (hunk 2 above)
Comment (giordano) [reply src/main/java/uk/ac/ucl/rc/development/oracc/ext/TextLineNumber.java]: I think so, the border looks right with this change, without it the border of the Arabic pane would be on the right.

File: src/main/java/uk/ac/ucl/rc/development/oracc/ext/TextLineNumber.java (hunk 1 above)
Comment (ageorgou) [reply src/main/java/uk/ac/ucl/rc/development/oracc/ext/TextLineNumber.java]: Ah, okay. The surrounding `JPanel` (the AtfAreaView) uses a `BorderLayout` manager, so it looks like we're okay just using `setPreferredSize`.

**Pull request merged**